### DISCUSSION TITLE
Contact List

### DISCUSSION THREAD

**Discussion by AnisTaluqdar** [syphon-org/syphon#562] [Ideas]
Every modern messaging app has a contact list.
I think we can create a private room and it can be used as a contact list as fluffychat did for their stories. 

The contact list can be 
1. Save by user manually 
2. Whenever people create a dm the contact can be saved automatically

The user ID can be saved as text in a private room and it can be shown as a contact list.

**Comment by notramo** [comment]
Why should it use a private room?
It is enough to list all the DMs, and gather the Matrix IDs from there.
Also, is it possible to save Matrix ID to the Android contact list?

**Reply by AnisTaluqdar** [reply]
> Why should it use a private room? 

Because we don't want to share our contact to others as a public room so we need a private room, hmm. also we have not any server side support for contact list so we can use private room.

**Reply by notramo** [reply]
Then why not write an MSC? This would make it possible to be compatible with other clients.

**Reply by AnisTaluqdar** [reply]
> Then why not write an MSC? This would make it possible to be compatible with other clients.

Hmm, They are working their way. But it's a long time 😑 still they didn't finished it.

**Reply by notramo** [reply]
Everybody can help writing and reviewing MSCs. If there is an existing MSC, then you can review it, and share your ideas.
If there isn't, you can write one.

**Comment by trymeouteh** [comment]
An MSC for a matrix account to store E2EE contact list woukd be amazing! With the ability to import and export your mateix contacts and have your matrix contacts synced between your devices within your mateix clients.

**Reply by notramo** [reply]
I guess it wouldn't be hard given that Server Side Secret Storage already exists. You just have to open an MSC specify which format the data should be stored.
